### PR TITLE
Quill locker/ create personal locker

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add functionality for staff to be able to create their own personal lockers on Quill Locker

## WHY
we want staff to be able to have an aesthetically pleasing way to organize any internal tooling or external links that they use in day to day work

## HOW
create several new components/utils/helper functions to handle the creation and updating of a personal locker

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2022-04-13 at 5 24 54 PM
Screen Shot 2022-04-13 at 5 37 20 PM
Screen Shot 2022-04-13 at 5 37 38 PM
Screen Shot 2022-04-13 at 5 37 57 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Add-functionality-to-create-personal-locker-3e3f872cff804d83977dd850cc558e39

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

File: services/QuillLMS/app/controllers/api/v1/lockers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class Api::V1::LockersController < Api::ApiController
+  def show
+    user_id = params[:id]
+    locker = Locker.where(user_id: user_id)&.first
```
Comment (emilia-friedberg) [inline services/QuillLMS/app/controllers/api/v1/lockers_controller.rb]: `Locker.find_by(user_id: user_id)` is this with fewer commands.

File: services/QuillLMS/app/controllers/api/v1/lockers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    if locker.save!
+      render json: { locker: locker }, status: 200
+    else
+      render json: locker.errors, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    user_id = params[:id]
+    locker = Locker.where(user_id: user_id)&.first
```
Comment (emilia-friedberg) [inline services/QuillLMS/app/controllers/api/v1/lockers_controller.rb]: See above.

**Review by emilia-friedberg — APPROVED** [review]
LGTM! Small Ruby comment. Love your use of constants!

**Review by cissyxyu — COMMENTED** [review]
Looks good. Excited to see this in action!

**Review by cissyxyu — APPROVED** [review]

**Pull request merged**